pkg/operator/status/status_controller_test.go

```diff
@@ -0,0 +1,289 @@
+				configv1.OperatorProgressing)
+			require.NotNil(t, progCond)
+			if test.expectProgressingTransition {
+				assert.True(t, progCond.LastTransitionTime.Time.After(
+					test.currentProgressingLastTransition.Time))
+			} else {
+				assert.Equal(t, test.currentProgressingLastTransition.Time.Format(time.UnixDate),
+					progCond.LastTransitionTime.Time.Format(time.UnixDate))
+			}
+		})
+	}
+}
+
+func TestConditionsEqual(t *testing.T) {
+	testCases := []struct {
+		description string
+		expected    bool
+		a, b        configv1.ClusterOperatorStatusCondition
+	}{
+		{
+			description: "empty conditions should be equal",
+			expected:    true,
+		},
+		{
+			description: "condition LastTransitionTime should be ignored",
+			expected:    true,
+			a: configv1.ClusterOperatorStatusCondition{
+				Type:               configv1.OperatorAvailable,
+				Status:             configv1.ConditionTrue,
+				LastTransitionTime: metav1.Unix(0, 0),
+			},
+			b: configv1.ClusterOperatorStatusCondition{
+				Type:               configv1.OperatorAvailable,
+				Status:             configv1.ConditionTrue,
+				LastTransitionTime: metav1.Unix(1, 0),
+			},
+		},
+		{
+			description: "check condition reason differs",
+			expected:    false,
+			a: configv1.ClusterOperatorStatusCondition{
+				Type:   configv1.OperatorAvailable,
+				Status: configv1.ConditionFalse,
+				Reason: "foo",
+			},
+			b: configv1.ClusterOperatorStatusCondition{
+
+				Type:   configv1.OperatorAvailable,
+				Status: configv1.ConditionFalse,
+				Reason: "bar",
+			},
+		},
+		{
+			description: "check condition message differs",
+			expected:    false,
+			a: configv1.ClusterOperatorStatusCondition{
+
+				Type:    configv1.OperatorAvailable,
+				Status:  configv1.ConditionFalse,
+				Message: "foo",
+			},
+
+			b: configv1.ClusterOperatorStatusCondition{
+
+				Type:    configv1.OperatorAvailable,
+				Status:  configv1.ConditionFalse,
+				Message: "bar",
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		actual := ConditionEqual(tc.a, tc.b)
+		if actual != tc.expected {
+			t.Fatalf("%q: expected %v, got %v", tc.description,
+				tc.expected, actual)
+		}
+	}
+}
+
+type expectedCondition struct {
+	conditionType configv1.ClusterStatusConditionType
+	reason        string
+	status        corev1.ConditionStatus
+}
+
+func TestConditions(t *testing.T) {
+	schemeutils.SetupScheme(scheme.Scheme)
+
+	tests := []struct {
+		name               string
+		expectedConditions []expectedCondition
+	}{
+		{
+			name: "set default conditions",
+			expectedConditions: []expectedCondition{
+				{
+					conditionType: configv1.OperatorAvailable,
+					status:        corev1.ConditionTrue,
+				},
+				{
+					conditionType: configv1.OperatorProgressing,
+					status:        corev1.ConditionFalse,
+				},
+				{
+					conditionType: configv1.OperatorDegraded,
+					status:        corev1.ConditionFalse,
+				},
+			},
+		},
+	}
```

Comment:
Coverage seems a little light in here, could we get something to simulate two handlers returning the same condition? Or is the handler registration making this infeasible?